lib/flash.c

```diff
@@ -125,6 +123,39 @@ uint8_t Flash_read_user_script( char* buffer )
     return 0;
 }
 
+char* Flash_script_name_from_mem( char* dest, char* src, int max_len )
+{
+    while( *src == '-' ){ src++; } // skip commments
+    while( *src == ' ' ){ src++; } // skip spaces
+    char* linebreak = strchr( src, '\n' );
+    if( linebreak ){ // print until newline
+        int len = (int)(linebreak - src);
+        if( len > max_len ){ len = max_len; }
+        strncpy( dest, src, len );
+    } else { // can't find a new line, so just fill the buffer
+        strncpy( dest, src, max_len );
+    }
+    return dest;
+}
+
+char* Flash_script_name( void )
+{
+    static char script[32];
+    memset( script, '\0', 32 );
+    switch( Flash_which_user_script() ){
+        case USERSCRIPT_Clear: strcpy( script, "User script cleared. Running." ); break;
```

Comment:
This is the case when there is no script?
If so wouldn't something like "No user script" be a clearer message? The "Running" at the end here is throwing me off, what is it running if there's no script?

And it seems these messages duplicate the ones here https://github.com/monome/crow/pull/227/files#diff-4c329923d1fdc04b8b83730be099f7dfR119
Could we merge them/the logic somehow?

Reply:
I agree it'd be nice to merge the logic.

Originally I had the same "No user script" message, but it's strange because it's what you see immediately after typing `^^c`, so it feels really strange. To me, responding to a user command with "No user script" *feels* like an error message, rather than confirmation of what the user just performed. I agree re: 'running'. will remove.

At present, the printing on boot isn't even working for some reason (an existing issue i had forgotten), so for now they are better described as the messages you see after changing what the user script is.

Reply:
Re: `^^c` I guess the only place/moment we can confirm to the user that the script has been cleared is immediately after clearing it (I guess that would be in `Flash_clear_user_script`).

I guess we could even omit passing the script name to `Lua_crowbegin` in that case?

Reply:
Updated the PR to consolidate the logic for 'print' and 'which script'.